rectangle.cpp

```diff
@@ -0,0 +1,75 @@
+#include "rectangle.h"
+
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+
+#include "indexbuffer.h"
+#include "renderer.h"
+#include "shader.h"
+#include "texture.h"
+#include "vertexarray.h"
+#include "vertexbuffer.h"
+
+static const unsigned int kIndicies[] = {
+    0, 1, 2,
+    2, 3, 0
+};
+
+Rectangle::Rectangle(const glm::vec3& position, float width, float height) :
+    mPosition(position),
+    mWidth(width),
+    mHeight(height),
+    mVB(4 * 4 * sizeof(float)),
+    mVA(),
```

Comment:
Is there a point to even do this, if these are empty?

Reply:
A couple suggestions here, hopefully this code example shows all of them:

```

// Use header file initialization, which means that your defaults are always filled in!
class Rectangle
{
private:
    float mWidth { 0.0f };
    float mHeight = 0.0f;  // equivalent (basically) to above
    std::unique_ptr<Texture *> mTexture { nullptr };
    
    SomeOtherClass {}; // Empty braces default initializes, should almost always be used if you just want your class to be initialized with all it's members
};


// all of the fields are initialized, so you don't need a default constructor, only initialize the unique things that you need for that specific constructor. In this case mTexture is nullptr!

// You can see how much boilerplate this reduces, and you can always trust your objects are initialized.

Rectangle :: Rectangle(float width) :
    mWidth { width }
    // height and texture are initialized
{
};


// Because we are using a unique ptr, we no longer need to delete the texture in the destructor
// destructor is defaulted!
// Rectangle :: ~Rectangle()
//{
    // delete texture 
//};

```

Reply:
Yeah that is much nicer. Is it better to initalize certain types like std::string with {} or "" or just leave them be like `std::string mFilePath;`. I assumed they would be left empty anyway, but might be good for consistency?

Reply:
It's completely up to you between the first two. I would initialize all of them for sure, just for consistency. The nice thing about "braced" or "universal" initialization is that you can be consistent with all types, and in general empty braces `{}` are well known to be "empty initialization". The benefit being you do not need to provide any additional information, just the empty braces.

so:
```
float mFloat {}; // inits to zero
float mAnotherFloat {0.0f}; // totally fine too, but the first one should be obvious

MyType mSomething {}; // assumed this is just some default or empty initialization

std::string mString {}; // can assume empty string
std::string mOther {""}; // more explicit
```


I generally like the braced version, but the equals version is good too. Here is a nice article about it:

https://blog.quasar.ai/2017/03/05/cpp-braced-initialization